src/utils/account/index.ts

```diff
@@ -66,9 +70,34 @@ const getAccountCredentials = (
   };
 };
 
-export const createAccount = (): AccountCredentials => {
+const getAccountCredentialsFromPem = (
+  pem: string,
+  subAccount?: number
+): AccountCredentialsFromPem => {
+  // Identity has boths keys via getKeyPair and PID via getPrincipal
+  const identity = Secp256k1KeyIdentity.fromPem(pem);
+  const accountId = getAccountId(identity.getPrincipal(), subAccount);
+  return {
+    pem,
+    identity,
+    accountId,
+  };
+};
+
+export const createAccount = (): AccountCredentialsFromMnemonic => {
   const mnemonic = bip39.generateMnemonic();
-  return getAccountCredentials(mnemonic, 0);
+  return getAccountCredentialsMnemonic(mnemonic, 0);
+};
+
+export const createAccountFromPem = (
+  pem: string,
+  accountId: number
+): AccountCredentials => {
+  if (!pem.length) throw new Error(ERRORS.INVALID_PEM);
+  if (typeof accountId !== 'number' || accountId < 0) {
```

Comment:
You might be interested in keeping the account validation reusable. As seen in L110.